mesa/experimental/jupyter_viz.py

```diff
@@ -9,48 +8,72 @@
 
 import mesa
 
-# Avoid interactive backend
-plt.switch_backend("agg")
-
-
-class JupyterContainer:
-    def __init__(
-        self,
-        model_class,
-        model_params,
-        measures=None,
-        name="Mesa Model",
-        agent_portrayal=None,
-    ):
-        self.model_class = model_class
-        self.split_model_params(model_params)
-        self.measures = measures
-        self.name = name
-        self.agent_portrayal = agent_portrayal
-        self.thread = None
-
-    def split_model_params(self, model_params):
-        self.model_params_input = {}
-        self.model_params_fixed = {}
-        for k, v in model_params.items():
-            if self.check_param_is_fixed(v):
-                self.model_params_fixed[k] = v
+
+@solara.component
+def JupyterViz(
```

Comment:
Yeah, this is simpler. The reason why I split into `JupyterContainer`, `MesaComponent`, and `JupyterViz` was the answer to these questions on Discord:
- https://discord.com/channels/1106593685241614489/1106593686223069309/1123620981911597208

The question:
> Is it possible to do just
```python
model_params = {
    "N": {
        "type": "SliderInt",
        "value": 50,
        "label": "Number of agents:",
        "min": 10,
        "max": 100,
        "step": 1,
    },
    "width": 10,
    "height": 10,
}
JupyterViz(
    BoltzmannWealthModel, model_params, measures=["Gini"], name="Money Model"
)
```

The answer,
> @rht you can also assign to page = MesaComponent(...) but they you would have to reverse your model, so JupyterViz shouldn't call MesaComponent, but the other way around

This was the reason for the `MesaComponent` and `JupyterViz` split. But in hindsight, the split wasn't necessary.

This was the reason for the `JupyterContainer` split: https://discord.com/channels/1106593685241614489/1106593686223069309/1123583246307954828.

Reply:
Yes, the power of hindsight... but still quite impressive to make it work with all the necessary basic features! Solara uses quite a different paradigm, so not everything is clear from the beginning